.github/workflows/codeql-analysis.yml

```diff
@@ -2,41 +2,44 @@ name: "CodeQL"
 
 on:
   push:
-    branches: [master]
+    branches: [ main ]
   pull_request:
     # The branches below must be a subset of the branches above
-    branches: [master]
+    branches: [ main ]
   schedule:
-    - cron: "30 0 * * 0"
+    - cron: '30 0 * * 0'
+    
+permissions:
+  contents: read
 
 jobs:
   analyze:
+    permissions:
+      security-events: write  # for github/codeql-action/autobuild to send a status report
     name: Analyze
     runs-on: ubuntu-latest
-    permissions:
-      actions: read
-      contents: read
-      security-events: write
 
     strategy:
       fail-fast: false
       matrix:
-        language: ["go"]
+        language: [ 'go' ]
 
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      # Initializes the CodeQL tools for scanning.
-      - name: Initialize CodeQL
-        uses: github/codeql-action/init@v1
-        with:
-          languages: ${{ matrix.language }}
+    - name: Checkout repository
+      uses: actions/checkout@v3
 
-      # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
-      # If this step fails, then you should remove it and run the build manually (see below)
-      - name: Autobuild
-        uses: github/codeql-action/autobuild@v1
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
```

Comment:
Thanks. Yes we thought about this before, but decided not to use it because we always try to pull in the latest ones as it includes additional new set of rules added by Github (like every week). It would be some considerable efforts to review this and update across all the repos on a timely basis.